src/structures/ClientApplication.js

```diff
@@ -11,35 +11,64 @@ class ClientApplication extends Application {
   _patch(data) {
     super._patch(data);
 
+    /**
+     * The flags this application has
+     * @type {number}
+     */
+    this.flags = data.flags ?? this.flags;
+
     /**
      * The app's cover image
      * @type {?string}
      */
-    this.cover = data.cover_image || null;
+    this.cover = data.cover_image ?? this.cover ?? null;
```

Comment:
i don't really like this, it would be better to keep it consistent with the other methods and set a value only if `'cover_image' in data`, the same goes for the rest of the properties

Reply:
sounds reasonable to me, how would i indicate that in jsdoc? add `{type|undefined}` everywhere or is it better to just leave a note at the class doc of `ClientApplication`